src/model/utils.rs

```diff
@@ -67,7 +67,7 @@ pub fn deserialize_members<'de, D: Deserializer<'de>>(
     let mut members = HashMap::new();
 
     for member in vec {
-        let user_id = member.user.read().id;
+        let user_id = futures::executor::block_on(member.user.read()).id;
```

Comment:
My bot is consistently deadlocking here.
I'm fairly new to Rust, but I'm getting here via `async fn recv_event` of the ShardRunner.
Isn't `block_on` in an asynchronous function considered to be very dangerous?

Reply:
```
tokio::task::block_in_place(|| { futures::executor::block_on(member.user.read()).id })
```
This "fixes" the deadlock but would be a rather ugly hack.
The exact same problem occurs quite a few times whenever you de-/serialize a tokio::RWLock and later need the ids for HashMaps.

Reply:
Yes, this is not good, indeed. I will look into this.

Reply:
Here is an insight into the issue: https://github.com/tokio-rs/tokio/issues/2376

Reply:
The idiomatic approach would certainly be an async serde-rs/json that doesnt exist (yet).
https://github.com/serde-rs/json/issues/575

The only option that I currently see is to skip the RWLocks during de-/serialization und introduce them afterwards.
Either by using a dedicated copy of the types for serde or by using a private field for each rwlock.

Reply:
Of course, `tokio::task::block_in_place` probably remains an unattractive, yet working solution.
For deserialization, `tokio::task::block_in_place` doesn't sound super evil since the objects haven't been shared with anyone yet.
For serialization, one could probably just deeply copy the object to make sure the read will never block as well.